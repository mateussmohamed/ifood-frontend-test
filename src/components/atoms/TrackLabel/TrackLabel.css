.track-label {
  flex-grow: 1;
  font-size: 14px;
}
