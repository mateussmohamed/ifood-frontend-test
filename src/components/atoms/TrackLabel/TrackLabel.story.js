import { storiesOf } from '@storybook/react';
import React from 'react';
import TrackLabel from './index';

const stories = storiesOf('TrackLabel', module);
stories.add('whit text prop', () => (
  <div style={{ float: 'left' }}>
    <TrackLabel text="Borders" />
  </div>
));
