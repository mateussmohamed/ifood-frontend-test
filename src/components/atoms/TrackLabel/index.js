import React from 'react';
import PropTypes from 'prop-types';
import './TrackLabel.css';

const TrackLabel = ({ text, type }) => (
  <span className={`track-label ${type}`}>{text}</span>
);

TrackLabel.defaultProps = {
  type: '',
};

TrackLabel.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string,
};

export default TrackLabel;
