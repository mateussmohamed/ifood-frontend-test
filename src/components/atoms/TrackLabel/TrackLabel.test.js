import React from 'react';
import renderer from 'react-test-renderer';
import TrackLabel from './index';

describe('TrackLabel', () => {
  test('should TrackLabel match to snapshot', () => {
    const component = renderer.create(<TrackLabel text="awesome text" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should when text prop render the correct text', () => {
    const component = renderer.create(<TrackLabel text="awesome text" />);
    expect(component.toTree().props.text).toBe('awesome text');
  });
});
