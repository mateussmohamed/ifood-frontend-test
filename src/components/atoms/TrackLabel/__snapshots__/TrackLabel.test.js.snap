// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackLabel should TrackLabel match to snapshot 1`] = `
<span
  className="track-label "
>
  awesome text
</span>
`;
